### DISCUSSION TITLE
Write Integrity

### DISCUSSION THREAD

**Discussion by robshep** [apache/iotdb#11841] [Q&A]
Hi,

Firstly, Thank you for the development of IoTDB. 🥇 

If a client `insert......(...)` succeeds, what assumptions can be made about the integrity of the data?

Are there any client-side API calls that can be made to help with discovering loss of integrity?

I'm wondering about the periodic use of `> FLUSH` can the client call this to discover any data problems?

In essence, I'm trying to add some client-side markers to help prevent silent data loss. 

Thanks for any pointers.

R

**Comment by HTHou** [comment]
Hi, if `insert......(...)` succeeds, IoTDB can make sure the integrity of the data inserted. Even through the data may be not flushed into 
the TsFile, the WAL will guarantee the data can be restored when IoTDB was shutdown and rebooted. 

The `FLUSH` command should used when you really hope that your data was written into the TsFile, which usually happened in the debug phase of development.
